mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageMapper.java

```diff
@@ -290,7 +290,7 @@ private Flux<UpdatedFlags> updatedFlags(List<ComposedMessageIdWithMetaData> list
                                             FlagsUpdateCalculator flagsUpdateCalculator) {
         return modSeqProvider.nextModSeqReactive(mailbox.getMailboxId())
             .flatMapMany(newModSeq -> Flux.fromIterable(listMessagesMetaData)
-                .flatMap(messageMetaData -> updateFlags(messageMetaData, flagsUpdateCalculator, newModSeq)));
+                .concatMap(messageMetaData -> updateFlags(messageMetaData, flagsUpdateCalculator, newModSeq)));
```

Comment:
Range updates are frequent in IMAP. doing this sequentially might be a show stopper.

How about using `flatMapSequential` ?

Also, we are missing the concurrency control: let's use the default concurrency defined in reactor utils.